pkg/services/accesscontrol/manager/evaluator.go

```diff
@@ -0,0 +1,72 @@
+package manager
+
+import (
+	"context"
+
+	"github.com/gobwas/glob"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+)
+
+func (ac *AccessControlService) Evaluate(ctx context.Context, user *models.SignedInUser, permission string, scope ...string) (bool, error) {
+	roles := []string{string(user.OrgRole)}
+	for _, role := range user.OrgRole.Children() {
+		roles = append(roles, string(role))
+	}
+	if user.IsGrafanaAdmin {
+		roles = append(roles, "Grafana Admin")
+	}
+
+	res, err := ac.GetUserPermissions(ctx, accesscontrol.GetUserPermissionsQuery{
+		OrgID:  user.OrgId,
+		UserID: user.UserId,
+		Roles:  roles,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	ok, dbScopes := extractPermission(res, permission)
+	if !ok {
+		return false, nil
+	}
+
+	for _, s := range scope {
+		var match bool
+		for dbScope := range dbScopes {
+			rule, err := glob.Compile(dbScope, ':', '/')
+			if err != nil {
+				return false, err
+			}
+
+			match = rule.Match(s)
+			if match {
+				break
+			}
+		}
+
+		if !match {
```

Comment:
Won't this always return false if there's no match found in first full iteration over `dbScopes`?

Reply:
Yes, right now it's defined as `scope AND scope AND scope AND scope`, so that should be correct, right?

Reply:
(fwiw, this entire file will be rewritten with a more reasonable API, this is just here to support early testing)